packages/cli/src/commonlib/azureLoginCI.ts

```diff
@@ -98,53 +154,155 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
     });
   }
 
-  public async setSubscriptionId(
-    subscriptionId: string,
-    root_folder = "./"
-  ): Promise<Result<null, FxError>> {
-    const token = await this.getAccountCredentialAsync();
-    const subscriptions = await this.getSubscriptionList(token!);
-
-    if (subscriptions.findIndex((sub) => sub.subscriptionId === subscriptionId) < 0) {
-      return err(NotFoundSubscriptionId());
-    }
-    AzureAccountManager.subscriptionId = subscriptionId;
-
-    /// TODO: use api's constant
-    const configPath = path.resolve(root_folder, `.${ConfigFolderName}/env.default.json`);
-    if (!(await fs.pathExists(configPath))) {
-      return err(NotSupportedProjectType());
+  async getStatus(): Promise<LoginStatus> {
+    await this.load();
+    if (
+      AzureAccountManager.clientId &&
+      AzureAccountManager.secret &&
+      AzureAccountManager.tenantId
+    ) {
+      return {
+        status: signedIn,
+      };
     }
-    const configJson = await fs.readJson(configPath);
-    configJson["solution"].subscriptionId = subscriptionId;
-    await fs.writeFile(configPath, JSON.stringify(configJson, null, 4));
-
-    return ok(null);
+    return {
+      status: signedOut,
+    };
   }
 
-  async getStatus(): Promise<LoginStatus> {
+  getJsonObject(showDialog?: boolean): Promise<Record<string, unknown> | undefined> {
     throw new Error("Method not implemented.");
   }
 
-  getJsonObject(showDialog?: boolean): Promise<Record<string, unknown> | undefined> {
-    throw new Error("Method not implemented.");
+  async listSubscriptions(): Promise<SubscriptionInfo[]> {
+    const credential = await this.getAccountCredentialAsync();
+    if (credential) {
+      const client = new SubscriptionClient(credential);
+      const tenants = await listAll(client.tenants, client.tenants.list());
```

Comment:
SP is bound in one tenant, why need to list tenants for SP?

Reply:
Good point, I have removed multi tenants part.